In [1]:
import requests
import matplotlib.pyplot as plt
import pandas as pd


In [2]:
endpoint = 'https://data.nashville.gov/resource/fuaa-r5cm.json'

In [3]:
response = requests.get(endpoint)

In [4]:
response

<Response [200]>

In [5]:
result = response.json()

In [6]:
search_dict = {}
for line in result:
    count = int(line.get('query_count'))
    text = line.get('query_text')
    if text in search_dict:
        search_dict[text] += count
    else:
        search_dict[text] = count

In [7]:
search_df = pd.DataFrame.from_dict(search_dict, orient = 'index')
search_df = search_df.reset_index()
search_df.columns = ['term', 'text']
search_df.head()

term  text
0           ebid   521
1  property maps   252
2           maps   237
3   property tax   214
4           jobs   210

In [8]:
search_df = search_df.sort_values('text', ascending=False)
search_df.head(10)

term  text
0                   ebid   521
1          property maps   252
2                   maps   237
10        property taxes   235
3           property tax   214
4                   jobs   210
5             employment   189
7                 police   167
8   criminal court clerk   167
9         media releases   165

### Question 1: Make an API request that returns the months where "fire" was searched in 2016. Which month had the most searches?

In [19]:
endpoint ='https://data.nashville.gov/resource/fuaa-r5cm.json'

In [20]:
args = '?year=2016&query_text=fire&$limit=50000'

In [21]:
fire = endpoint + args

In [22]:
response = requests.get(fire)

In [23]:
response

<Response [200]>

In [24]:
fire_df = pd.read_json(fire)

In [25]:
fire_df = fire_df.reset_index(drop= True)

In [26]:
fire_df = fire_df.sort_values('query_count',ascending = False)

In [27]:
fire_df

month_name  year  query_count query_text
7      August  2016           47       fire
9     October  2016           38       fire
11   December  2016           38       fire
8   September  2016           36       fire
1    February  2016           35       fire
2       March  2016           32       fire
10   November  2016           32       fire
5        June  2016           31       fire
3       April  2016           26       fire
4         May  2016           24       fire
6        July  2016           24       fire
0     January  2016           19       fire

#### Fire was searched the most in August

### Question 2: Make an API request that returns all the times a query was run more than 100 times in a month. How many times did this occur?

In [36]:
endpoint ='https://data.nashville.gov/resource/fuaa-r5cm.json'

In [37]:
args = '?$where=query_count>100&$limit=10000'

In [38]:
query_large = endpoint + args

In [39]:
response = requests.get(query_large)

In [40]:
result = response.json()
len(result)

1262

In [41]:
query_large_df = pd.read_json(query_large)

In [42]:
query_large_df

month_name  year  query_count            query_text  month
0         March  2014          101               permits    NaN
1       January  2015          101  criminal court clerk    NaN
2     September  2015          101                 codes    NaN
3         March  2016          101                police    NaN
4         March  2016          101         civil service    NaN
...         ...   ...          ...                   ...    ...
1257  September  2018          750     annual enrollment    NaN
1258    October  2018          816     annual enrollment    NaN
1259    January  2019         2646             Nashville    1.0
1260  September  2019         5327             directory    9.0
1261  September  2019         5610                  test    9.0

[1262 rows x 5 columns]

####  A query was run more than 100 times in a month 1,262 times

### Question 3: Make another API request that returns all the times "codes" was searched more than 100 times in a month. How many times did this occur?

In [43]:
endpoint ='https://data.nashville.gov/resource/fuaa-r5cm.json'

In [44]:
args = '?$where=query_count>100&query_text=codes&$limit=10000'

In [45]:
big_codes = endpoint + args

In [46]:
response = requests.get(big_codes)

In [47]:
result = response.json()
len(result)

56